src/generator.ts

```diff
@@ -1014,15 +1011,26 @@ export class Generator {
   async install(
     install?: string | Install | (string | Install)[]
   ): Promise<void | { staticDeps: string[]; dynamicDeps: string[] }> {
-    if (arguments.length > 1)
-      throw new JspmError(
-        "Install takes no arguments, a single install target, or a list of install targets."
-      );
+    return this._install(install);
+  }
+
+  private async _install(
```

Comment:
Not sure I fully understand the reason for this private `_install` function?

Reply:
Ah, I missed that we were keeping `mode` part of the private API. Should we perhaps just expose it though with defaults? Is there a risk in doing that?